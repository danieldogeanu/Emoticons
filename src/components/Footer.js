import React, { Component } from 'react';
import Link from '../elements/Link';
import '../styles/components/Footer.scss';

class Footer extends Component {
	constructor() {
		super();
		this.state = {
			author: { name: 'Daniel Dogeanu', url: 'https://danieldogeanu.com' },
			social: [
				{ name: 'Twitter', url: 'https://twitter.com/danieldogeanu' },
				{ name: 'GitHub', url: 'https://github.com/danieldogeanu' },
				{ name: 'Behance', url: 'https://behance.net/danieldogeanu' },
			]
		};
	}

	render() {
		const {author, social} = this.state;
		return (
			<footer className="Footer">
				<p className="author">By: <Link data={author} /></p>
				<p className="mynets">
					<span className="folw">Follow Me: </span>
					<span className="wrapper">
						{social.map((value) => {
							return <Link key={value.name.toLowerCase()} data={value} social={true} />;
						})}
					</span>
				</p>
			</footer>
		);
	}
}

export default Footer;
