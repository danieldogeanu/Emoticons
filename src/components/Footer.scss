@import '../styles/includes';

.Footer {
	display: flex;
	width: 100%;

	p {
		display: block;
		width: 50%;
		font-size: rem(13);
		font-weight: weight(light);
		text-transform: uppercase;
	}

	a {
		transition: color 0.3s ease;
		color: color-scheme(item, btn);
		&:hover { color: color-scheme(item, btn-hover); }
	}

	.author {
		text-align: left;
	}

	.mynets {
		text-align: right;
	}
}