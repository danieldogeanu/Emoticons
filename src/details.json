{
	"labels": [
		{
			"class": "face",
			"name": "Face"
		},
		{
			"class": "name",
			"name": "Emoticon Name"
		},
		{
			"class": "copy",
			"name": "Get"
		}
	],
	"author": {
		"name": "Daniel Dogeanu",
		"url": "https://danieldogeanu.com"
	},
	"social": [
		{
			"name": "Twitter",
			"url": "https://twitter.com/danieldogeanu"
		},
		{
			"name": "GitHub",
			"url": "https://github.com/danieldogeanu"
		},
		{
			"name": "LinkedIn",
			"url": "https://www.linkedin.com/in/danieldogeanu/"
		},
		{
			"name": "Behance",
			"url": "https://behance.net/danieldogeanu"
		}
	],
	"shortcuts": [
		{
			"key": "s",
			"description": "Selects Search Input"
		},
		{
			"key": "esc",
			"description": "Clears Search Input / Exits Menu"
		},
		{
			"key": "k",
			"description": "Opens Keyboard Shortcuts Screen"
		},
		{
			"key": "t",
			"description": "Scrolls Back Up"
		}
	]
}
