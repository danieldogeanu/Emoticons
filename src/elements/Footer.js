import React from 'react';
import Link from './Link';
import { author, social } from '../details.json';
import '../styles/components/Footer.scss';

const Footer = () => (
	<footer className="Footer" data-testid="Footer">
		<p className="author" data-testid="author">
			By: <Link data={author} />
		</p>
		<p className="mynets" data-testid="mynets">
			<span className="folw">Follow Me: </span>
			<span className="wrapper" data-testid="wrapper">
				{social.map((value) => (
					<Link key={value.name.toLowerCase()}
								data={value} social={true} />
				))}
			</span>
		</p>
	</footer>
);

export default Footer;
