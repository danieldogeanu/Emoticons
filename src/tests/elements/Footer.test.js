import React from 'react';
import {render} from '@testing-library/react';
import {author, social} from '../../details.json';
import Footer from '../../elements/Footer';

describe('Footer Element', () => {

	it('renders footer element properly', () => {
		const {container} = render(<Footer />);
		const renderedFooter = container.querySelector('footer');

		expect(renderedFooter).toBeInTheDocument();
		expect(renderedFooter).toHaveClass('Footer');
		expect(renderedFooter).toMatchSnapshot();
	});

	it('contains author section', () => {
		const {getByTestId} = render(<Footer />);
		const renderedAuthor = getByTestId('author');
		const renderedLink = renderedAuthor.querySelector('a');

		expect(renderedAuthor).toBeInTheDocument();
		expect(renderedAuthor).toHaveClass('author');
		expect(renderedAuthor).toHaveTextContent('By: ' + author.name);
		expect(renderedAuthor).toContainElement(renderedLink);
		expect(renderedLink).toHaveAttribute('href', author.url);
	});

	it('contains social section', () => {
		const netsClass = 'mynets';
		const {getByTestId, getByText, getAllByTestId} = render(<Footer />);
		const renderedNets = getByTestId(netsClass);
		const renderedFollow = getByText(/follow me/i);
		const renderedWrapper = getByTestId('wrapper');
		const renderedLinks = getAllByTestId('mynet');

		expect(renderedNets).toBeInTheDocument();
		expect(renderedNets).toHaveClass(netsClass);
		expect(renderedNets).toContainElement(renderedFollow);
		expect(renderedNets).toContainElement(renderedWrapper);

		expect(renderedFollow).toHaveClass('folw');
		expect(renderedFollow).toHaveTextContent('Follow Me:');

		expect(renderedWrapper).toHaveClass('wrapper');
		expect(renderedLinks.length).toBe(4);
	});

	it('contains all social networks', () => {
		const {getAllByTestId} = render(<Footer />);
		const renderedNets = getAllByTestId('mynet');

		renderedNets.forEach((net, i) => {
			expect(net).toHaveTextContent(social[i].name);
		});
	});

});
