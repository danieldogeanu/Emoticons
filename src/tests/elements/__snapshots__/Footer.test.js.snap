// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Element renders footer element properly 1`] = `
<footer
  class="Footer"
>
  <p
    class="author"
    data-testid="author"
  >
    By: 
    <a
      data-testid="Link"
      href="https://danieldogeanu.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Daniel Dogeanu
    </a>
  </p>
  <p
    class="mynets"
    data-testid="mynets"
  >
    <span
      class="folw"
    >
      Follow Me: 
    </span>
    <span
      class="wrapper"
      data-testid="wrapper"
    >
      <a
        data-testid="Link"
        href="https://twitter.com/danieldogeanu"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="mynet"
          data-testid="container"
        >
          <span
            class="show-for-screen-reader"
          >
            Twitter
          </span>
          <svg
            class="icon twitter"
          >
            <use
              xlink:href="#twitter"
            />
          </svg>
        </span>
      </a>
      <a
        data-testid="Link"
        href="https://github.com/danieldogeanu"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="mynet"
          data-testid="container"
        >
          <span
            class="show-for-screen-reader"
          >
            GitHub
          </span>
          <svg
            class="icon github"
          >
            <use
              xlink:href="#github"
            />
          </svg>
        </span>
      </a>
      <a
        data-testid="Link"
        href="https://www.linkedin.com/in/danieldogeanu/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="mynet"
          data-testid="container"
        >
          <span
            class="show-for-screen-reader"
          >
            LinkedIn
          </span>
          <svg
            class="icon linkedin"
          >
            <use
              xlink:href="#linkedin"
            />
          </svg>
        </span>
      </a>
      <a
        data-testid="Link"
        href="https://behance.net/danieldogeanu"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="mynet"
          data-testid="container"
        >
          <span
            class="show-for-screen-reader"
          >
            Behance
          </span>
          <svg
            class="icon behance"
          >
            <use
              xlink:href="#behance"
            />
          </svg>
        </span>
      </a>
    </span>
  </p>
</footer>
`;
